src/login.rs

```diff
@@ -1,3 +1,5 @@
+use adw::prelude::MessageDialogExtManual;
+use adw::traits::MessageDialogExt;
```

Comment:
I think it's cleaner if you change `use gtk::prelude::*;` to `use adw::prelude::*;` instead of adding these two lines.